frontend/src/containers/RegisterPage/RegisterPage.js

```diff
@@ -1,31 +1,81 @@
 import { Grid } from '@material-ui/core';
 import React from 'react';
+import { register } from '../../api/PublicAPI';
+import CustomSnackbar from '../../components/CustomSnackbar/CustomSnackbar';
 import RegistrationForm from '../../components/RegistrationForm/RegistrationForm';
 
 class RegisterPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showGeneralError: false,
+      showUserExistsError: false,
+    };
+  }
 
-  // [TM]: TODO WDB-28 also delete line below
-  // eslint-disable-next-line no-unused-vars 
   handleRegister = (firstName, lastName, email, password) => {
-    // Call to BE
-    // After successful registration reroute to confirm email
-    const { history } = this.props;
-    history.push('/confirm-email');
-  }
+    const data = {
+      firstName,
+      lastName,
+      email,
+      password,
+    };
+    register(data)
+      .then(() => {
+        const { history } = this.props;
+        history.push('/confirm-email');
+      })
+      .catch((ex) => {
+        try {
+          const { status } = ex.response;
+          if (status === 409) this.setState({ showUserExistsError: true });
+          else this.setState({ showGeneralError: true });
+        } catch {
+          this.setState({ showGeneralError: true });
+        }
+      });
+  };
 
   render() {
+    const { showGeneralError, showUserExistsError } = this.state;
+
+    const hideGeneralError = () => {
+      this.setState({ showGeneralError: false });
+    };
+
+    const hideUserExistsError = () => {
+      this.setState({ showUserExistsError: false });
+    };
     return (
-      <Grid
-        container
-        className="root"
-        direction="column"
-        alignItems="center"
-        justify="center"
-      >
-        <Grid item xs={10} sm={6} md={4}>
-          <RegistrationForm onRegister={this.handleRegister} />
+      <>
+        {showGeneralError && (
+          <CustomSnackbar
+            topCenter
+            message="Bad request"
```

Comment:
You should probably say "A server error has occurred. Try again later" or something similar here. "Bad Request" is not very informative. Especially, f.e. if the server is down, the user would just get "Bad Request" and be dumbfounded about what the hell did Aurimas think when writing this here error message.

Reply:
Yeah, fuck the regular User man. I would say to him: "Get your shit together and figure out server error responses"